### ISSUE TITLE
Asus Strix 18 (2023) 4090 - Dynamic Boost is not working

### ISSUE THREAD

**Issue by Frub3L** [seerge/g-helper#1663]
## NOTE
Bug reports without clear information or scenario to reproduce and logs from ``%AppData%\GHelper`` will be closed without answer. 
Please respect the time of the developer. Thanks.

**Describe the bug**
Not sure if that's a bug, but I was not able to make my GPU use the dynamic boost, it constantly power throttles at 150w (it's supposed to be 175w with DB. 

**Expected behavior**
Changing the Dynamic Boost slider also doesn't seem to change anything. Neither Turbo or Balanced mode seem to make a difference as well. 

**App Logs**
11/28/2023 5:27:27 PM: Mode = 0 : OK
11/28/2023 5:27:27 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:27 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:27 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:27 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:27 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:27 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:28 PM: PowerLimit A0 = 150 : OK
11/28/2023 5:27:28 PM: PowerLimit A3 = 150 : OK
11/28/2023 5:27:29 PM: PowerLimit C0 = 25 : OK
11/28/2023 5:27:29 PM: PowerLimit C2 = 87 : OK
11/28/2023 5:27:32 PM: PowerLimit A0 = 125 : OK
11/28/2023 5:27:32 PM: PowerLimit A3 = 125 : OK
11/28/2023 5:27:32 PM: PowerLimit C0 = 25 : OK
11/28/2023 5:27:32 PM: PowerLimit C2 = 87 : OK
11/28/2023 5:27:36 PM: PowerLimit C0 = 24 : OK
11/28/2023 5:27:36 PM: PowerLimit C2 = 87 : OK
11/28/2023 5:27:36 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:36 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:36 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:36 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:37 PM: PowerLimit C0 = 25 : OK
11/28/2023 5:27:37 PM: PowerLimit C2 = 87 : OK
11/28/2023 5:27:37 PM: GET CLOCK LIMIT: 0
11/28/2023 5:27:37 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:37 PM: GET GPU CLOCKS: 0, 0
11/28/2023 5:27:37 PM: GET CLOCK LIMIT: 0

**Screenshots or screencasts**

![Screenshot 2023-11-28 172807](https://github.com/seerge/g-helper/assets/139390959/b3908726-5dc2-40bf-b92a-70c5160d46e7)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 11] - WINDOWS 11 installed
 - Laptop model - Asus Strix 18 (2023) G834JY

**Asus software**
- Armoury Crate (or it's services installed) - Armourcy Crate is uninstalled, services are installed but disabled.
- MyASUS installed - Yes.
- Other Asus services running in background. - No

**Additional context**
None

**Comment by seerge** [comment]
@Frub3L how do you test that in a first place ? Try to run Furmark, it is known to utilize as much of GPU power as possible.

Next to that make sure that you have latest and **clean** install of nvidia drivers (preferably after [DDU](https://www.guru3d.com/download/display-driver-uninstaller-download/))

**Comment by Frub3L** [comment]
@seerge Hi! thanks for the reply. I have tested my laptop in many games, including benchmarks, I also tested it in furmark as well, and it won't go past 150w and constantly power throttle. 

I have installed clean drivers after wiping them out with DDU (I always do that). Yes, I was tinkering with the Dynamic Boost slider, because I thought that it might be bugged and I just had to "refresh" the value, but nothing worked, including the Turbo mode. 

I was only able to go past 150w when I rolled back my nvidia drivers from the latest to 528.XX, and set it manually in MSI Afterburner (that feature is disabled in the latest nvidia drivers). Do you have any idea why it happens? 
Thanks!

**Comment by seerge** [comment]
@Frub3L I don't know why it happens.

Dynamic boost slider in G-Helper is the same as in AC, and it's a shortcut to corresponding endpoint in bios that lets you limit max power on that side.

If GPU is actually being able to get more power with older driver, it means that bios is not restricting it. So I would assume it's some software issue. What does HWinfo or similar apps show when GPU is fully loaded in furmark. Is it hitting power limits or may be temp / voltage limits ?

Anyway, there is nothing I can do, as whole boost "command" is dead simple, and same for all models. You can run it even in powershell (as admin)


```
(Get-WmiObject -Namespace Root/WMI -Class AsusAtkWmi_WMNB).DEVS(0x001200C0, 25)
````

**Comment by Frub3L** [comment]
HWinfo basically shows that it's constantly power throttling, it hovers at around 145-150W at all times in any game. It can't be a temperature limit since it barely reaches 78 Celsius. What would you suggest then? I am almost certainly sure that it has to be a software bug or something. BTW when I play on Ultimate GPU mode for example Red Dead Redemption 2 is literally unplayable, even tho I reach above 100fps the game has insanely bad and unstable frame pacing. Does that ring a bell with something? It is very weird since this is the only game that has these issues (on standard mode it's perfectly fine). In general Ultimate GPU mode seems to have issues with having stable Core clocks as well (I didn't undervolt my GPU yet, only my CPU).

**Comment by seerge** [comment]
@Frub3L well, if games are unplayable on 4090 , i don't even know what to say :) as for me looks like it's some driver issues

May be try DDU again and clean all that shaders caches, etc :)

Ultimate mode has nothing to do with clocks. It's just a toggle for MUX switch (i.e. your internal display is routed directly to dGPU), it doesn't have any other effects

**Comment by DLdota** [comment]
+25 watts is a big lie, it can be seen only in synthetic tests (u can check). 
i also cant increase on my Strix 17 (2023) to 140 watts in games, only 115.

**Comment by seerge** [comment]
@DLdota well, for 4060/4070 it's generally difficult to get 140W indeed, but when I had my Strix 16 4080 it was easily running 175W and with something like +300/+1000 core/memory offsets :)

**Comment by Frub3L** [comment]
> @Frub3L well, I'm afraid I can't help you anyhow here. May be try to do a fresh windows instlall if that bothers you so much ? :|

Thank you for the help, yes I wanted to avoid reinstalling my windows completely, but I guess I have no other choice lol

**Comment by Frub3L** [comment]
> +25 watts is a big lie, it can be seen only in synthetic tests (u can check). i also cant increase on my Strix 17 (2023) to 140 watts in games, only 115.

It is actually not a lie. Notice what I said before, I was able to go over 150w only after installing older Nvidia drivers and manually changing the power limit (they unlock the slider in MSI Afterburner). The games had no issues going above 150w. On the other hand using the latest drivers it seems like 150w is hard-locked no matter what I do and what mode I use. 
@seerge raised an important observation that since I was able to manually change the power limit and go over 150w on the older driver, that mean it is not locked in bios. Based on that I am almost certainly sure it has to be related to some kind of software bug.

**Comment by Frub3L** [comment]
Okay, so I have found a fix, it was rather weird, but it works...
Some time ago, I read online that the "Nvidia Platform Controllers and Framework" actually controls your dynamic boost. I just remembered it now and checked it. Turns out I had an error there (yellow triangle with Error code 19 if I am not mistaken). I tried disabling and reinstalling it, but it didn't work.

WHAT WORKED:
![image](https://github.com/seerge/g-helper/assets/139390959/3fb827dd-6f33-4fe4-a037-66d4231266e2)
So what I basically did was to roll back to the previous version of the driver, using the option in the properties section, and restart my laptop. After the restart, the yellow triangle was gone, BUT it was not fixed yet. What helped me was to --> Update the driver--> Search automatically for drivers (then it installed the newer version) --> Device restart. After restarting and enabling turbo mode my games go above 150w with no problem.
![image](https://github.com/seerge/g-helper/assets/139390959/eb0bba08-15e9-4f88-82be-34eb0d663f29)

I don't know if you ever heard of this fix for that problem but feel free to add it somewhere in the fixes/bugs section if you have any. Anyway thanks for the help @seerge !
